crates/walrus-service/src/node/errors.rs

```diff
@@ -411,6 +411,10 @@ pub enum DbCheckpointError {
     CheckpointCreationError(String),
     #[error("Backup already in progress")]
     CheckpointInProgress,
+    #[error("No checkpoint found")]
+    NoCheckpointFound,
+    #[error("Restore error: {0}")]
+    RestoreError(String),
```

Comment:
Nit: maybe we can do `RestoreError(#[from] rocksdb::Error),` to preserve the error struct instead of using a string.